borg/archiver.py

```diff
@@ -177,6 +178,7 @@ def do_check(self, args, repository):
     @with_repository()
     def do_change_passphrase(self, args, repository, manifest, key):
         """Change repository key file passphrase"""
+        check_repository_compatibility(manifest, REPO_OPERATION_CHECK)
```

Comment:
I made a version using a `compatibility` argument to `with_repository`
Do you like this version better? I made it very explicit about locations where there is no checking.